factories/resolverFactory.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+/*jslint latedef:false*/
+
+(function() {
+  angular.module('o19s.splainer-search')
+    .factory('ResolverFactory', [
+      '$q',
+      'searchSvc',
+      'solrUrlSvc',
+      'normalDocsSvc',
+      ResolverFactory
+    ]);
+
+  function ResolverFactory($q, searchSvc, solrUrlSvc, normalDocsSvc) {
+    var Resolver = function(ids, settings, chunkSize) {
+      var self        = this;
+
+      self.settings   = settings;
+      self.ids        = ids;
+      self.docs       = [];
+      self.args       = {};
+      self.config     = {};
+      self.queryText  = null;
+      self.fieldSpec  = self.settings.createFieldSpec();
+      self.chunkSize  = chunkSize;
+
+      self.fetchDocs  = fetchDocs;
+
+      if ( self.settings.searchEngine === undefined || self.settings.searchEngine === 'solr' ) {
+        var escapeIds = function(ids) {
+          var newIds = [];
+          angular.forEach(ids, function(id) {
+            newIds.push(solrUrlSvc.escapeUserQuery(id));
+          });
+          return newIds;
+        };
+
+        var allIdsLuceneQuery = self.fieldSpec.id + ':(';
+        allIdsLuceneQuery += escapeIds(ids).join(' OR ');
+        allIdsLuceneQuery += ')';
+        self.queryText = allIdsLuceneQuery;
+
+        self.args = {
+          defType: ['lucene'],
+          rows: [ids.length],
+          q: ['#$query##']
+        };
+      } else if ( settings.searchEngine === 'es' ) {
+        self.args = {
+          'query': {
+            'ids': {
+              'values': ids
+            }
+          }
+        };
+      }
+
+      self.config = {
+        sanitize:   false,
+        highlight:  false,
+        debug:      false,
+      };
+
+      self.searcher = searchSvc.createSearcher(
+        self.fieldSpec.fieldList(),
+        self.settings.searchUrl,
+        self.args,
+        self.queryText,
+        self.config,
+        self.settings.searchEngine
+      );
+
+      function fetchDocs () {
```

Comment:
Does this interfere at all with the client-side doc caching done in Quepid? I can't remember how the docCache svc there is tied to this (maybe its just external).